### PR TITLE
5091 lang attribuutti

### PR DESCRIPTION
**Pull request by TeroOtavia**
- Added lang attribute to workspace description
- Added lang attribute to workspace materials
- Added lang attribute to material editor
- Added lang attribute to material loader components
- Added lang attribute to content topic and element
- Lang attribute by default comes from workspace settings which defaults
to "fi" if not set.
- Lang attribute prioritizing order material_page > material_section > workspace

Resolves: #5091

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/workspace/WorkspaceJsPluginMessages_en.properties [hunk 1]
```diff
@@ -230,6 +230,8 @@ plugin.workspace.materialsManagement.editorView.subTitle.producers = Producers
 plugin.workspace.materialsManagement.editorView.addProducers.placeHolder = Add Producer
 plugin.workspace.materialsManagement.editorView.subTitle.license = License
 plugin.workspace.materialsManagement.editorView.license.none = No license
+plugin.workspace.materialsManagement.editorView.subTitle.localeCode = Locale code
```
Comment (aviljakaine) [inline muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/workspace/WorkspaceJsPluginMessages_en.properties:233]: Lines mismatch with the fi locale

**Review by aviljakaine — COMMENTED** [review]
Conflicts need to be fixed.
Backend seems ok if the null valued parameters are checked to be as desired.

**Review by aviljakaine — APPROVED** [review]
Backend ok

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/WorkspaceMaterialController.java [hunk 2]
```diff
@@ -922,7 +926,7 @@ private ContentNode createContentNode(WorkspaceNode rootMaterialNode, int level,
           rootMaterialNode.getId(), null, level, null, null, rootMaterialNode.getParent().getId(),
           nextSibling == null ? null : nextSibling.getId(), rootMaterialNode.getHidden(), null,
           workspaceFolder.getPath(), null, Collections.emptyList(), folderViewRestrict, 
-          false);
+          false, workspaceFolder.getLanguage() == null ? null : workspaceFolder.getLanguage());
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/WorkspaceMaterialController.java:929]: Could simply say `workspaceFolder.getLanguage()`

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java [hunk 3]
```diff
@@ -1101,6 +1102,10 @@ public Response updateWorkspace(@PathParam ("WORKSPACEENTITYID") Long workspaceE
     if (payload.getPublished() != null && !workspaceEntity.getPublished().equals(payload.getPublished())) {
       workspaceEntity = workspaceEntityController.updatePublished(workspaceEntity, payload.getPublished());
     }
+    
+    if (payload.getLanguage() != null && !workspaceEntity.getLanguage().equals(payload.getLanguage())) {
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java:1106]: NPE chance if workspace entity language is null

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceSystemRESTService.java [hunk 4]
```diff
@@ -449,7 +449,7 @@ public Response relocateHelpPages(@Context Request request) {
         
         // ...then create a new section under the help folder and move all pages under it
         
-        WorkspaceFolder helpSection = workspaceMaterialController.createWorkspaceFolder(helpRoot, "Suoritusohjeet", "suoritusohjeet", 0);
+        WorkspaceFolder helpSection = workspaceMaterialController.createWorkspaceFolder(helpRoot, "Suoritusohjeet", "suoritusohjeet", null);
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceSystemRESTService.java:452]: `WorkspaceLanguage.fi`

File: updates/1.127/221212_Lang_attribute.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pkukkon) [inline updates/1.127/221212_Lang_attribute.xml:1]: Needs version bump

**Review by pkukkon — CHANGES_REQUESTED** [review]

**Review by pkukkon — APPROVED** [review]

**Review by aviljakaine — APPROVED** [review]
Backend ok

**Review by avukkonen — APPROVED** [review]

**Pull request merged**